### ISSUE TITLE
CoreStoreObject snapshot doesn't work with ObjectPublisher

### ISSUE THREAD

**Issue by joeljfischer** [JohnEstropia/CoreStore#364]
Referencing a property on an `ObjectSnapshot<MyObject>` where `MyObject` inherits from `CoreStoreObject` results in an error:

> Referencing subscript 'subscript(dynamicMember:)' on 'ObjectSnapshot' requires that 'MyObject' inherit from 'NSManagedObject'

Do I have to change my object to be an `NSManagedObject` to get this to work?

**Comment by JohnEstropia** [comment]
It simply means the signature doesn't match. Can you show the code you are getting this error in?

**Comment by joeljfischer** [comment]
Sure, I'll try to provide the relevant code:

```swift
// In a preparation method of a table view cell
observer = object.asPublisher(in: CoreStoreDefaults.dataStack)
observer?.addObserver(self, { [weak self] (publisher) in
    self?.currentSnapshot = publisher.snapshot
    if let snapshot = self?.currentSnapshot {
        self?.updateUI(snapshot: snapshot)
    }
})

private func updateUI(snapshot: ObjectSnapshot<MyObject>) {
    if snapshot.isPaused { // Accessing a property of MyObject fails with the error noted in the issue description
        playPauseButton.setImage(UIImage(systemName: "play.circle.fill"), for: .normal)
    } else {
        playPauseButton.setImage(UIImage(systemName: "pause.circle.fill"), for: .normal)
    }
}
```

**Comment by JohnEstropia** [comment]
What is `isPaused`? Is it a computed property? If so you will need to write that property in an extension:
```swift
extension ObjectSnapshot where O == MyObject {
    var isPaused: Bool {
        // ...
    }
}
```

**Comment by joeljfischer** [comment]
That's a good catch. It is a computed property, I didn't think about that. So I will need to write the computed property twice, once on an `ObjectSnapshot` extensions and once on the actual object itself (if I need to access it in both of those contexts)?

**Comment by JohnEstropia** [comment]
Unfortunately that's a language limitation for now.

There will be a new way to do this using `@Field.Virtual` in an upcoming version (Swift 5.2): https://github.com/JohnEstropia/CoreStore/issues/359
It will be released soon after Xcode 11.4 goes out of beta.

**Comment by joeljfischer** [comment]
Thanks for the help, this issue can be closed if you wish.
